core/controllers/question_editor.py

```diff
@@ -25,12 +25,47 @@
 import feconf
 
 
+class QuestionCreationHandler(base.BaseHandler):
+    """A handler that creates the question model given a question dict."""
+
+    @acl_decorators.can_manage_question_skill_status
+    def get(self):
+        """Handles GET requests."""
+        if not feconf.ENABLE_NEW_STRUCTURES:
+            raise self.PageNotFoundException
+
+        new_question_id = question_services.get_new_question_id()
+        self.render_json({
+            'questionId': new_question_id
+        })
+
+    @acl_decorators.can_manage_question_skill_status
+    def post(self):
+        """Handles POST requests."""
+        if not feconf.ENABLE_NEW_STRUCTURES:
+            raise self.PageNotFoundException
+
+        question_dict = self.payload.get('question_dict')
+        if (
+                ('id' not in question_dict) or
+                ('question_state_data' not in question_dict) or
+                ('language_code' not in question_dict)):
+            raise self.InvalidInputException
+
+        question_dict['version'] = 1
+        question_dict['question_state_schema_version'] = (
+            feconf.CURRENT_EXPLORATION_STATES_SCHEMA_VERSION)
+        question = question_domain.Question.from_dict(question_dict)
```

Comment:
Is there any sort of validation step in this flow? If not, should there be?

Bear in mind that you can receive practically anything from the frontend. So, do careful validation to ensure that anything that makes its way further into the backend is completely safe.

Reply:
Yeah, there is a validation is _creatte_new_question(), just before actually saving, though it was not strict. Will make this a strict validation.

Reply:
Done.